lib/scholar/linear/bayesian_ridge_regression.ex

```diff
@@ -0,0 +1,320 @@
+      type:
+        {:or,
+         [
+           {:custom, Scholar.Options, :non_negative_number, []},
+           {:list, {:custom, Scholar.Options, :non_negative_number, []}},
+           {:custom, Scholar.Options, :weights, []}
+         ]},
+      doc: """
+      The weights for each observation. If not provided,
+      all observations are assigned equal weight.
+      """
+    ],
+    fit_intercept?: [
+      type: :boolean,
+      default: true,
+      doc: """
+      If set to `true`, a model will fit the intercept. Otherwise,
+      the intercept is set to `0.0`. The intercept is an independent term
+      in a linear model. Specifically, it is the expected mean value
+      of targets for a zero-vector on input.
+      """
+    ],
+    alpha_init: [
+      type: {:custom, Scholar.Options, :non_negative_number, []},
+      doc: ~S"""
+      The initial value for alpha. This parameter influences the precision of the noise.
+      `:alpha` must be a non-negative float i.e. in [0, inf).
+      Defaults to 1/Var(y).      
+      """
+    ],
+    lambda_init: [
+      type: {:custom, Scholar.Options, :non_negative_number, []},
+      default: 1.0,
+      doc: ~S"""
+      The initial value for lambda. This parameter influences the precision of the weights.
+      `:lambda` must be a non-negative float i.e. in [0, inf).
+      Defaults to 1.
+      """
+    ],
+    alpha_1: [
+      type: {:custom, Scholar.Options, :non_negative_number, []},
+      default: 1.0e-6,
+      doc: ~S"""
+      Hyper-parameter : shape parameter for the Gamma distribution prior      
+      over the alpha parameter.
+      """
+    ],
+    alpha_2: [
+      type: {:custom, Scholar.Options, :non_negative_number, []},
+      default: 1.0e-6,
+      doc: ~S"""
+      Hyper-parameter : inverse scale (rate) parameter for the Gamma distribution prior
+      over the alpha parameter.
+      """
+    ],
+    lambda_1: [
+      type: {:custom, Scholar.Options, :non_negative_number, []},
+      default: 1.0e-6,
+      doc: ~S"""
+      Hyper-parameter : shape parameter for the Gamma distribution prior
+      over the lambda parameter.            
+      """
+    ],
+    lambda_2: [
+      type: {:custom, Scholar.Options, :non_negative_number, []},
+      default: 1.0e-6,
+      doc: ~S"""
+      Hyper-parameter : inverse scale (rate) parameter for the Gamma distribution prior      
+      over the lambda parameter.            
+      """
+    ],
+    eps: [
+      type: :float,
+      default: 1.0e-8,
+      doc:
+        "The convergence tolerance. When `Nx.sum(Nx.abs(coef - coef_new)) < :eps`, the algorithm is considered to have converged."
+    ]
+  ]
+
+  @opts_schema NimbleOptions.new!(opts)
+  deftransform fit(x, y, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @opts_schema)
+
+    opts =
+      [
+        sample_weights_flag: opts[:sample_weights] != nil
+      ] ++
+        opts
+
+    {sample_weights, opts} = Keyword.pop(opts, :sample_weights, 1.0)
```

Comment:
I would set default `sample_weights` to `Nx.broadcast(Nx.as_type(1.0, x_type), {num_samples})`.